modyn/config/schema/pipeline.py

```diff
@@ -128,14 +131,10 @@ class MultiDownsamplingConfig(BaseModel):
     )
 
     @model_validator(mode="after")
-    @classmethod
-    def validate_downsampling_thresholds(
-        cls,
-        data: "MultiDownsamplingConfig",
-    ) -> "MultiDownsamplingConfig":
-        if len(data.downsampling_thresholds) != len(data.downsampling_list) - 1:
+    def validate_downsampling_thresholds(self) -> Self:
+        if len(self.downsampling_thresholds) != len(self.downsampling_list) - 1:
             raise ValueError("The downsampling_thresholds list should have one less item than the downsampling_list.")
-        return data
+        return self
 
```

Comment:
@robinholzi We should add tests against non-trivial validations because I found some not-working code...
Here, when `model_validator` is used with `after` mode, it is passed with an instance of the class.
See https://docs.pydantic.dev/latest/concepts/validators/#model-validators

Reply:
Personally, I wouldn't add tests for the validation as it will fail loudly when sth. is wrong in almost all cases. Then one can simply adjust the model (e.g. if a field type is off). The same applies to the usage of @field_validator and @model_validator. But I agree, if there's complex logic inside the custom validation functions it makes sense to write a small unit test though

Reply:
I didn't get your argument? You mean we cannot find out which file/test function fails in the `pytest` output because there are too many errors? I think the summary is easy to analyze.

I believe it's worth adding unit tests to these custom validation functions, otherwise we might make mistakes without being aware of it (for example here)